### DISCUSSION TITLE
android_SDK

### DISCUSSION THREAD

**Discussion by mspakdel** [beeware/beeware#166] [General]
hello 
in tutorial 5 on windows when i am executing briefcase create android command it gives me an error whch is "You did not accept the Android SDK licenses. Please re-run the briefcase command
and accept the Android SDK license when prompted. You may need an Internet
connection"
how can i solve this problem after executing that code 
there is no folder name android in src folder of my project

**Comment by mhsmith** [comment]
Were there any other messages about licenses?

If you haven't already, try upgrading to the current Briefcase version, 0.3.9, which has some improvements in this area, and also produces more detailed log files.

If that version still gives the same error, please find the log file, whose location will be printed just before Briefcase exits, and post it here by dragging it into the GitHub comment box.

**Reply by mspakdel** [reply]
[briefcase.2022_08_30-18_13_54.create.log](https://github.com/beeware/beeware/files/9453342/briefcase.2022_08_30-18_13_54.create.log)

**Reply by mspakdel** [reply]
there is no prompt at all when running "briefcase create android" command

**Reply by mspakdel** [reply]
![Screenshot 2022-08-30 182950](https://user-images.githubusercontent.com/31448714/187457068-f7ec4868-9c14-4bb4-bde9-069cd519a2f2.jpg)
this is also the CLI output screenshot

**Reply by freakboy3742** [reply]
Thanks for those details; FWIW, screenshots of terminal sessions are rarely very helpful - the *text* content of the console might be helpful, but a screenshot should be considered the last resort for providing console output. That's why we asked for the log file - the log file contains *all* the history of what was printed to the console, along with other details, in a searchable form. 

Two more follow up questions:

1. How/where are you running this script? In particular, are you running it in a context where stdin might be disabled (e.g., inside an IDE which is "read only")?
2. Does the folder `C:\Users\TP-Main\AppData\Local\BeeWare\briefcase\Cache\tools\android_sdk/licenses` exist? If so, what file are contained in it?

**Reply by mspakdel** [reply]
sorry for being late response and also thanks for helping figure it out
for your first question i am running it inside vscode and didn't know its read only or not
for your second question no there is not a licenses folder and it isn't created by default.  there is just one folder inside of android_sdk and its name is: cmdline-tools.

**Reply by mhsmith** [reply]
> for your first question i am running it inside vscode and didn't know its read only or not

OK, then it's worth trying it again in a Command Prompt window. But with the warnings in your screenshot, it seems more likely that sdkmanager is having trouble accessing the internet.

Do you access the internet through a proxy? If so, try setting the HTTP_PROXY environment variable, which sdkmanager [allegedly respects](https://issuetracker.google.com/issues/79122663). Or if that doesn't work, take the sdkmanager command from the log file, and run it manually while adding the proxy-related arguments shown [here](https://developer.android.com/studio/command-line/sdkmanager).
